src/StoreApi/SessionHandler.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Automattic\WooCommerce\StoreApi;
+
+use Automattic\Jetpack\Constants;
+use Automattic\WooCommerce\StoreApi\Utilities\JsonWebToken;
+use WC_Session;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * SessionHandler class
+ */
+final class SessionHandler extends WC_Session {
+	/**
+	 * Token from HTTP headers.
+	 *
+	 * @var string
+	 */
+	protected $token;
+
+	/**
+	 * Table name for session data.
+	 *
+	 * @var string Custom session table name
+	 */
+	protected $_table; // phpcs:ignore PSR2.Classes.PropertyDeclaration.Underscore
+
+	/**
+	 * Expiration timestamp.
+	 *
+	 * @var int
+	 */
+	protected $_session_expiration; // phpcs:ignore PSR2.Classes.PropertyDeclaration.Underscore
+
+	/**
+	 * Constructor for the session class.
+	 */
+	public function __construct() {
+		$this->token  = wc_clean( wp_unslash( $_SERVER['HTTP_CART_TOKEN'] ?? '' ) );
+		$this->_table = $GLOBALS['wpdb']->prefix . 'woocommerce_sessions';
+	}
+
+	/**
+	 * Init hooks and session data.
+	 */
+	public function init() {
+		$this->init_session_from_token();
+		add_action( 'shutdown', array( $this, 'save_data' ), 20 );
+	}
+
+	/**
+	 * Process the token header to load the correct session.
+	 */
+	protected function init_session_from_token() {
+		$payload = JsonWebToken::get_parts( $this->token )->payload;
+
+		$this->_customer_id        = $payload->user_id;
+		$this->_session_expiration = $payload->exp;
+		$this->_data               = (array) $this->get_session( $this->_customer_id, array() );
+	}
+
+	/**
+	 * Returns the session.
+	 *
+	 * @param string $customer_id Customer ID.
+	 * @param mixed  $default Default session value.
+	 *
+	 * @return string|array|bool
+	 */
+	public function get_session( $customer_id, $default = false ) {
+		global $wpdb;
+
+		if ( Constants::is_defined( 'WP_SETUP_CONFIG' ) ) {
+			return false;
+		}
+
+		$value = $wpdb->get_var(
+			$wpdb->prepare(
```

Comment:
This is causing an SQL error. If error reporting/logging is enabled you should see it when you make the request. The table name cannot be prepared because WordPress will wrap it in quotes.

Reply:
Fixed 👍🏼